### DISCUSSION TITLE
Transfer data from Parent to Child

### DISCUSSION THREAD

**Discussion by patriciozeus** [serenity-is/Serenity#7043] [Help]
Hello Community

I need to transfer data from
MasterDetailRelation. I attach an image and the reference code :

TaBieArriendoRow
.....
.....
[DisplayName("Tipo Contacto"), Column("arr_tipo_contacto")]
public ComboTipoContacto? ArrTipoContacto
{
get { return (ComboTipoContacto)Fields.ArrTipoContacto[this]; }
set { Fields.ArrTipoContacto[this] = (Int32)value; }
}
....
//////// MONTOS ARRIENDO (VINCULACION)
[DisplayName("Monto Arriendo"), MasterDetailRelation(foreignKey: "ape_arriendo", MasterKeyField = "arr_numero", FilterValue = "TaBieArriendoPersonasRow",IncludeColumns = "ArrTipoContacto"), NotMapped]
public List DetalleArriendo
{
get { return Fields.DetalleArriendo[this]; }
set { Fields.DetalleArriendo[this] = value; }
}
....
....
public RowListField DetalleArriendo;

TaBieArriendoPersonasRow
....
...
[DisplayName("Tipo Contacto"), Column("ape_tipo_contacto")]
public ComboTipoContacto? ApeTipoContacto
{
get { return (ComboTipoContacto)Fields.ApeTipoContacto[this]; }
set { Fields.ApeTipoContacto[this] = (Int32)value; }
}
...
...

TaBieArriendoPersonasEditor
....
....
protected validateEntity(row: TaBieArriendoPersonasRow, id: number) {
if (!super.validateEntity(row, id))
return false;
row.ArrNumero = Q.toId(row.ArrNumero);

        //
        var sameProduct = Q.tryFirst(this.view.getItems(), x => x.ApeTipoMonto === row.ApeTipoMonto);
        if (sameProduct && this.id(sameProduct) !== id) {
            Q.alert('Este tipo de Monto ya existe en el detalle del Arriendo !!');
            return false;
        }
        //

        row.TmoGlosa = TaBieArriendoTipoMontoRow.getLookup().itemById[row.ApeTipoMonto].TmoGlosa;
        row.ApeMonto = TaBieArriendoTipoMontoRow.getLookup().itemById[row.ApeTipoMonto].TmoMonto;
        row.ApeMontoTotal = row.ApeMonto * row.ApeCantidad;

        return true;
    }

![ayuda](https://github.com/serenity-is/Serenity/assets/154239047/d724a071-6be1-4314-8548-05230135cfea)

**Comment by nahidulislam-cse15** [comment]
---child dialog.ts- 
// first declare a public property value in child
  public masterId: string;
//after load entity set the parent value to the form
 if (this.isNew() && this.masterId != null) {
            // FormSerialNo - field we want set 
this.form.FormSerialNo.value = this.masterId;
            this.LoadAppInfo();
            
        }
-- parent dialog.ts-
//  set child dilog value then call the dialog-
//AdmissionMoneyReceiptDialog()- child
	
	var dlg = new AdmissionMoneyReceiptDialog();
 
	dlg.masterId = this.form.FormSerialNo.value;
        
  	dlg.loadNewAndOpenDialog();

**Reply by patriciozeus** [reply]
Hello dear

I have a question regarding the following instruction:
this.LoadAppInfo();

At what point do you define it?

Thanks for your help.

Atte.-

**Comment by nahidulislam-cse15** [comment]
this.LoadAppInfo(); is need not to be  define for your project .i load info using the passing value
